logstash-core/src/main/java/org/logstash/config/ir/compiler/AbstractFilterDelegatorExt.java

```diff
@@ -131,7 +130,7 @@ public RubyArray multiFilter(final IRubyObject input) {
     @SuppressWarnings("rawtypes")
     public RubyArray flush(final IRubyObject input) {
         RubyHash options = (RubyHash) input;
-        final ThreadContext context = WorkerLoop.THREAD_CONTEXT.get();
+        final ThreadContext context = options.getRuntime().getCurrentContext();
```

Comment:
why don't we do it consistently using `RubyUtil.RUBY.getCurrentContext()` ?

Reply:
no preference on my end, just very used to taking runtime from local state.
(keeping the runtime static has often been an anti-pattern for JRuby exts)

Reply:
every JRuby ext object is multi-runtime capable and constructed with a `runtime` passed.
so in most cases it makes sense to use the runtime it got, instead of a 'global' one ...
(theoretically we could leverage that in tests - booting a runtime and throwing it away).

Reply:
Ok got it - thanks. Then the opposite question 😆 should we then consistently use local state everywhere?

Reply:
:yin_yang: would be preferable ... but I think its fine to rather just be consistent for the moment.
if its not needed I certainly do not see the point of pushing this, could naively imagine that it might have been useful functional testing ruby pipeline configurations (which changed the runtime state?) but that is deprecated these days :speaking_head:

Reply:
Up to you to decide! In any case will LGTM!